### PR TITLE
Fixup iceberg GitHub Action workflow

### PR DESCRIPTION
**Pull request by kennknowles**
There was an interpolation variable incidentally left in it when I forked from another file.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by Abacn** [comment]
Let's also add this workflow to https://github.com/apache/beam/blob/master/.github/workflows/README.md so there is an eye on the status of it

The naming was inherited from Jenkins job names. I understand it may be verbose and not ideal, but until we get it all clean up I would suggest keep the naming convention (beam_PreCommit_Java_Iceberg_IO_Direct.yml)

File: .github/workflows/IO_Iceberg.yml [hunk 1]
```diff
@@ -71,8 +71,7 @@ jobs:
       github.event_name == 'push' ||
       github.event_name == 'pull_request_target' ||
       (github.event_name == 'schedule' && github.repository == 'apache/beam') ||
-      github.event_name == 'workflow_dispatch' ||
-      github.event.comment.body == 'Run Java_Amqp_IO_Direct PreCommit'
```
Comment (Abacn) [inline .github/workflows/IO_Iceberg.yml:75]: We can keep this. The phrase trigger for PreCommit is currently working

Comment (kennknowles) [reply .github/workflows/IO_Iceberg.yml:75]: Done

**Comment by kennknowles** [comment]
https://github.com/apache/beam/actions/runs/8650999727/job/23720718775 shows that it works and finds a problem in the module

**Review by Abacn — APPROVED** [review]

**Pull request merged**